Source/DafnyDriver/DafnyDoc.cs

```diff
@@ -0,0 +1,1142 @@
+      }
+
+      String link = Link(null, name, name);
+      String mss = ReplaceFirst(ms, m.Name, link);
+
+      summaries.Append(mss);
+      if (!String.IsNullOrEmpty(docstring)) {
+        summaries.Append(space4).Append(DashShortDocstring(md));
+      }
+      summaries.Append(br).Append(eol);
+
+      details.Append(Anchor(name)).Append(eol);
+      details.Append(RuleWithText(name)).Append(eol);
+      if (!String.IsNullOrEmpty(modifiers)) {
+        details.Append(modifiers).Append(br).Append(eol);
+      }
+      details.Append(m.WhatKind).Append(br).Append(eol);
+      mss = ReplaceFirst(ms, m.Name, Bold(name));
+      details.Append(mss).Append(br).Append(eol);
+
+      var attributes = m.Attributes?.ToString();
+      if (!String.IsNullOrEmpty(attributes)) {
+        details.Append(space4).Append(attributes).Append(br).Append(eol);
+      }
+      details.Append(IndentedHtml(docstring));
+      AppendSpecs(details, m);
+    }
+  }
+
+  public void WriteLemmas(TopLevelDeclWithMembers decl, StringBuilder summaries, StringBuilder details) {
+    var methods = decl.Members.Where(m => m is Method && (m as Method).IsLemmaLike).Select(m => m as MemberDecl).ToList();
+    methods.Sort((f, ff) => f.Name.CompareTo(ff.Name));
+    if (methods.Count() > 0) {
+      summaries.Append(Heading3("Lemmas")).Append(eol);
+      details.Append(Heading3("Lemmas")).Append(eol);
+      WriteMethodsList(methods, decl, summaries, details);
+    }
+  }
+
+  public string IndentedHtml(string docstring) {
+    return IndentedHtml(docstring, false);
+  }
+
+  public string IndentedHtml(string docstring, bool nothingIfNull) {
+    if (!String.IsNullOrEmpty(docstring)) {
+      return Indent(ToHtml(docstring)) + eol;
+    } else if (!nothingIfNull) {
+      return br + eol;
+    } else {
+      return "";
+    }
+  }
+
+  // returns true iff some specs were appended to the StringBuilder
+  public bool AppendSpecs(StringBuilder details, MemberDecl d) {
+    bool some = false;
+    if (d is Method) {
+      var m = d as Method;
+      foreach (var req in m.Req) {
+        details.Append(space4).Append("<b>requires</b> ").Append(req.E.ToString()).Append(br).Append(eol);
+        some = true;
+      }
+      if (m.Mod != null && m.Mod.Expressions.Count > 0) {
+        var list = String.Join(", ", m.Mod.Expressions.Select(e => e.OriginalExpression.ToString() + (e.FieldName != null ? "`" + e.FieldName : "")));
+        details.Append(space4).Append(Bold("modifies")).Append(" ").Append(list).Append(br).Append(eol);
+        some = true;
+      }
+      foreach (var en in m.Ens) {
+        details.Append(space4).Append("<b>ensures</b> ").Append(en.E.ToString()).Append(br).Append(eol);
+        some = true;
+      }
+      if (m.Decreases != null && m.Decreases.Expressions.Count > 0) {
+        var dec = String.Join(", ", m.Decreases.Expressions.Select(e => e.ToString()));
+        details.Append(space4).Append("<b>decreases</b> ").Append(dec).Append(br).Append(eol);
```

Comment:
Could you please make sure this style for ensures and decreases is not hard-coded? i would prefer a span with a class that sets font-weight: bold.